core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -67,4 +67,19 @@ public String getName() {
   public Collection<MethodDescriptor<?, ?>> getMethods() {
     return methods;
   }
+
+  @Override
+  public boolean equals(Object other) {
+    if (!(other instanceof ServiceDescriptor)) {
+      return false;
+    }
+    ServiceDescriptor that = (ServiceDescriptor) other;
+    return name.equals(that.name)
+        && methods.equals(that.methods);
```

Comment:
Normalized.  

It seems ServiceDescriptor should probably have been more stringent, else equality would have been easier to define.